AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOrchestrator.swift

```diff
@@ -138,7 +126,17 @@ final class AWSInitialSyncOrchestrator: InitialSyncOrchestrator {
         initialSyncOrchestratorTopic.send(value)
     }
 
-    private func onReceiveCompletion() {
+    private func onReceiveCompletion(modelSchema: ModelSchema, result: Subscribers.Completion<DataStoreError>) {
+        if case .failure(let dataStoreError) = result {
+            let syncError = DataStoreError.sync(
+                "An error occurred syncing \(modelSchema.name)",
+                "",
+                dataStoreError)
+            self.syncErrors.append(syncError)
+        }
+
```

Comment:
is this logic no longer needed?
```
if self.isUnauthorizedError(syncError) {
      self.initialSyncOrchestratorTopic.send(.finished(modelName: modelSchema.name))
}
```

Reply:
yup, it's no longer needed because we will always mark the initial sync as finished regardless of the error. we can and should in the future reuse the method to adhere to the `errorHandler` behavior, ie.

```
if self.isUnauthorizedError(syncError) {
   self.errorHandler(SyncError.unauthorizedError)
}
```

I left this out to isolate the changes in this PR to be just related to the consistency of emitting the ModelSyncedEvent, and kept the method since i anticipated that we'll be using it soon in a future change